servicetalk-opentelemetry/src/main/java/io/servicetalk/opentelemetry/RequestTagExtractor.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright © 2022 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.servicetalk.opentelemetry;
+
+import io.servicetalk.http.api.StreamingHttpRequest;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+
+final class RequestTagExtractor implements TagExtractor<StreamingHttpRequest> {
```

Comment:
Doesn't look like access to the payload body is required from this extractor. Consider using `HttpRequestMetaData` instead of `StreamingHttpRequest` then.